docs/storage.md

```diff
@@ -129,3 +129,37 @@ Note that on the read path, Prometheus only fetches raw series data for a set of
 ### Existing integrations
 
 To learn more about existing integrations with remote storage systems, see the [Integrations documentation](https://prometheus.io/docs/operating/integrations/#remote-endpoints-and-storage).
+
+## Backfilling for OpenMetrics format
+
+### Overview
+
+If a user wants to migrate from an existing monitoring system (source) to Prometheus, they can add their older data to TSDB easily using backfilling. The source system may be Prometheus or another TSDB which is capable of dumping data in [OpenMetrics](https://openmetrics.io/) format. 
+
+Sample OpenMetrics file:
+
+```
+# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{code="200"} 1021 1565133713.989
+http_requests_total{code="200"} 1 1565133714.989
+http_requests_total{code="400"} 2 1565133715.989
+# EOF
+```
+
+Backfilling is implemented by doing multiple passes through the OpenMetrics file and processing a 2-hour block at a time. OpenMetrics input file is read separately for each future block to only read the lines that belong into the respective block. This ensures Prometheus keeps at most 2 hours of data in the memory.
+
+### Usage 
+
+Backfilling can be used via the promtool command line. This tool will create all Prometheus blocks, in a temporary workspace. The blocks might be read later by prometheus and compacted, so you should make sure that the blocks belong to the prometheus users.
+
+By default temp workspace is data/, you can change it by passing the name of the desired output directory.
+
+```
+./promtool create-blocks-from openmetrics name_of_input_file name_of_output_file
+```
+
+If there is an overlap, you will need to shut down Prometheus, and restart it with `--storage.tsdb.allow-overlapping-blocks`. If there is no overlap or if the flag is already set, no shut-down is required. 
+
+Do not forget the `--storage.tsdb.retention.time=X` flag, if you not want to lose any imported data points.
```

Comment:
What do you mean by gotcha ? And what it the gotcha ?

Reply:
A gotcha is something that makes complete sense in its context, but which would surprise users.

For example here --storage.tsdb.retention.time is completely unrelated to this tool, however it seems likely it will surprise a fair few users so it's appropriate to give a quick warning.

Reply:
Sorry, I still don't understand what you want me to do here. I have mentioned the gotcha

Reply:
My comment still applies, this is incorrect and misleading. The warning should be to keep in mind that data older than the retention will be deleted.